table/block_based/filter_policy.cc

```diff
@@ -1525,9 +1537,20 @@ BloomLikeFilterPolicy::GetStandard128RibbonBuilderWithContext(
     const FilterBuildingContext& context) const {
   // FIXME: code duplication with GetFastLocalBloomBuilderWithContext
   bool offm = context.table_options.optimize_filters_for_memory;
-  bool reserve_filter_construction_mem =
-      (context.table_options.reserve_table_builder_memory &&
-       context.table_options.block_cache);
+  // bool reserve_filter_construction_mem =
+  //     (context.table_options.block_cache &&
+  //      context.table_options.cache_usage_options
+  //              .options_overrides[static_cast<uint32_t>(
+  //                  CacheEntryRole::kFilterConstruction)]
+  //              .charged == CacheEntryRoleOptions::Decision::kEnabled);
+  const auto iter =
+      context.table_options.cache_usage_options.options_overrides.find(
+          CacheEntryRole::kFilterConstruction);
+  const auto& reserve_filter_construction_mem =
+      iter != context.table_options.cache_usage_options.options_overrides.end()
+          ? iter->second.charged == CacheEntryRoleOptions::Decision::kEnabled
+          : context.table_options.cache_usage_options.default_options.charged ==
+                CacheEntryRoleOptions::Decision::kEnabled;
```

Comment:
This expression scares me, originally because of its length/density and later because it appears to be assigning a temporary boolean to a reference. I guess the `const bool&`'s lifetime will be extended and it's indeed correct but it is too much extra difficulty to understand. Can we add an if-else and remove the reference?

Reply:
Fixed by comparing .charged with CacheEntryRoleOptions::Decision instead of bool.